ci-operator/step-registry/gather/core-dump/gather-core-dump-commands.sh

```diff
@@ -12,5 +12,19 @@ echo "Gathering node core dumps ..."
 mkdir -p ${ARTIFACT_DIR}/core-dumps
 
 oc adm must-gather --dest-dir="${ARTIFACT_DIR}/core-dumps" -- sh -c "/usr/bin/gather_core_dumps || true"
+
+ls -R ${ARTIFACT_DIR}/core-dumps
+find ${ARTIFACT_DIR}/core-dumps -type f
+num_core_dumps=$(find ${ARTIFACT_DIR}/core-dumps/*/ -type f | wc -l)
+
 tar -czC "${ARTIFACT_DIR}/core-dumps" -f "${ARTIFACT_DIR}/core.dumps.tar.gz" .
 rm -rf "${ARTIFACT_DIR}/core-dumps"
+
+# if there are any files collected from the core-dump gather and $FAIL_ON_CORE_DUMP is true then exit 1
+# so the job can see it and be marked as a failure
+if [ "${FAIL_ON_CORE_DUMP}" == "true" ]; then
+  if [ $num_core_dumps -ne 0 ]; then
+    echo "Fail: Found core dump files."
+    exit 1
```

Comment:
If a POST step in the workflow fails and exits with exitcode 1 does it actually fail the entire Job? (I would think not)

Reply:
yes it does. here is an example:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/20566/rehearse-20566-periodic-ci-openshift-release-master-ci-4.9-upgrade-from-stable-4.8-e2e-aws-ovn-upgrade/1420980093785739264